src/alire/alire-toml_index.adb

```diff
@@ -343,6 +343,48 @@ package body Alire.TOML_Index is
       Dirs.End_Search (Search);
    end Load_Catalog;
 
+   ----------------------------
+   -- Load_Release_From_File --
+   ----------------------------
+
+   function Load_Release_From_File
+     (Filename    : String;
+      Environment : Environment_Variables)
+      return        Releases.Release
+   is
+      Pkg      : Package_Type;
+      Result   : Load_Result;
+      Value    : TOML.TOML_Value;
+      Releases : Containers.Release_Sets.Set;
+   begin
+      Value := Load_TOML_From_File (Filename, Result);
+      if not Result.Success then
+         raise Constraint_Error with "could not load " & Filename;
+      end if;
+
+      --  Convert it to our intermediate data structures
+
+      Decode_TOML_Package
+        (Filename, Ada.Directories.Base_Name (Filename), Environment, Value, Pkg, Result);
+      if not Result.Success then
+         raise Constraint_Error with +Result.Message;
+      end if;
+
+      --  Generate the releases to be imported
+
+      Import_TOML_Package_As_Releases (Pkg, Environment, Releases);
+
+      --  Verify and return
```

Comment:
These checks happen after we have imported the releases in the in-memory index, so this procedure kind of corrupts the state in case of error. To fix this, what about moving these checks before the call to `Import_TOML_Packages_As_Releases`? (checking `Pkg.Releases` instead of `Releases`)

Reply:
The intention is that Import_TOML_Packages_As_Releases no longer imports into the in-memory index (the Register call is moved to the new Index_Releases, lines 503-507). So Load_Release_From_File doesn't touch the in-memory index. (I'll verify that.)

If the Import_* in the procedure name is confusing I could rename to Convert_TOML_Package_To_Releases or something like that, if you prefer.

Reply:
Renamed to Decode_TOML_Package_To_Releases

Reply:
Looks good, thanks!